### ISSUE TITLE
Initalization failure with Lua 5.3

### ISSUE THREAD

**Issue by hishamhm** [LuaLanes/lanes#111]
I'm writing some Lua 5.3 code and I needed threads, so my first option was Lanes :)

First I needed to edit locally the `lanes-scm-1.rockspec` to edit the Lua version dependency. Probably time for a scm-2 revision in the repo? 

On to the issue. The latest code built fine with Lua 5.3, but failed initializing:

```
lua: /Users/hisham/luarocks/prefix-5.3/share/lua/5.3/lanes.lua:326: can't copy non-deep full userdata across lanes
stack traceback:
	[C]: in function 'lanes.core.lane_new'
	/Users/hisham/luarocks/prefix-5.3/share/lua/5.3/lanes.lua:326: in function </Users/hisham/luarocks/prefix-5.3/share/lua/5.3/lanes.lua:324>
	/Users/hisham/luarocks/prefix-5.3/share/lua/5.3/lanes.lua:569: in function 'lanes.configure'
	../../test.lua:2: in main chunk
	[C]: in ?
```

(In spite of the /Users/ directory, this is on [Linux](http://www.gobolinux.org).)

Then I decided to run `luarocks unpack` to take a look at the code, tweaked tools.h to enable USE_DEBUG_SPEW, ran `luarocks make` and got this, after lots of debug messages:

```
--- KEY: 'io'
--- KEY: 'stderr'
--- USERDATA
lua: /Users/hisham/luarocks/prefix-5.3/share/lua/5.3/lanes.lua:326: can't copy non-deep full userdata across lanes
```

Looks like it's failing when propagating io.stderr. I managed to initialize lanes by passing `demote_full_userdata` on `lanes.configure()`. Not sure if things will work from here on, but I'll report back if I find any surprises!

**Comment by benoit-germain** [comment]
I don't think that `demote_full_userdata` will do the trick, as it simply passes the object as a light userdata instead, so it's unlikely that code will work on the receiving end.

**Comment by benoit-germain** [comment]
I think this is fixed by 3e3605dde3dd304fdce50171963c5363b31e1115.
